streams/src/test/java/org/apache/kafka/streams/state/internals/CachingPersistentWindowStoreTest.java

```diff
@@ -119,29 +125,24 @@ public void closeStore() {
         cachingStore.close();
     }
 
-    @SuppressWarnings("deprecation")
+    @SuppressWarnings({"deprecation", "unchecked"})
     @Test
     public void shouldDelegateDeprecatedInit() {
-        final WindowStore<Bytes, byte[]> inner = EasyMock.mock(WindowStore.class);
+        final WindowStore<Bytes, byte[]> inner = mock(WindowStore.class);
```

Comment:
You could move the mock specification to the object fields and use `@Mock` to avoid suppression. Same is true for other places in this PR where suppression is applied.

Reply:
In these situations we sometimes use a mock and sometimes the real thing. For example, in most test cases we use the real thing
```
@Before
public void setUp() {
    underlyingStore = new RocksDBWindowStore(bytesStore, false, WINDOW_SIZE);
    ...
    cachingStore = new CachingWindowStore(underlyingStore, WINDOW_SIZE, SEGMENT_INTERVAL);
}
```
but for the test you picked we want a mock and defining it inline seems a bit cleaner.